Sources/SPMBuildCore/BinaryTarget+Extensions.swift

```diff
@@ -46,7 +46,7 @@ extension BinaryTarget {
         // Filter the libraries that are relevant to the triple.
         // FIXME: this filter needs to become more sophisticated
         guard let library = metadata.libraries.first(where: {
-            $0.platform == triple.os.asXCFrameworkPlatformString &&
+            ($0.platform, $0.platformVariant) == triple.asXCFrameworkPlatformString ?? ("", "") &&
```

Comment:
I think you want:

```swift
($0.platform, $0.platformVariant ?? "") == triple.asXCFrameworkPlatformString ?? ("", "")
```

The extra `?? ""` is needed because `nil != ""` and the absence of a value is equivalent to a nil value for our purposes here.